lib/api/objectDelete.js

```diff
@@ -125,12 +122,19 @@ export default function objectDelete(authInfo, request, log, cb) {
         const resHeaders = collectCorsHeaders(request.headers.origin,
             request.method, bucketMD);
         if (err) {
-            if (err === skipError) {
-                return cb(null, resHeaders);
+            if (err !== skipError) {
+                log.debug('error processing request', { error: err,
+                    method: 'objectDelete' });
+                return cb(err, resHeaders);
             }
-            log.debug('error processing request', { error: err,
-                method: 'objectDelete' });
-            return cb(err, resHeaders);
+            log.trace('received request to delete non-existing version');
+            if (reqVersionId) {
+                resHeaders['x-amz-version-id'] = reqVersionId === 'null' ?
+                    reqVersionId : versionIdUtils.encode(reqVersionId);
+            }
+            pushMetric('deleteObject', log, { authInfo, bucket: bucketName,
```

Comment:
I checked with Rahul and he said we don't increment the counter for the api if the error is NoSuchKey or NoSuchVersion. I removed the pushMetric call here.